docs/asciidoc/kotlin.adoc

```diff
@@ -0,0 +1,104 @@
+[[kotlin]]
+= Kotlin support
+
+[[kotlin-introduction]]
+https://kotlinlang.org[Kotlin] is a statically-typed language targeting the JVM (and other platforms),
+which allows writing concise and elegant code while providing very good
+https://kotlinlang.org/docs/reference/java-interop.html[interoperability] with
+existing libraries written in Java.
+
+This section describes Reactor's support for Kotlin.
+
+[[kotlin-requirements]]
+== Requirements
+
+Reactor supports Kotlin 1.1+ and requires
+https://bintray.com/bintray/jcenter/org.jetbrains.kotlin%3Akotlin-stdlib[`kotlin-stdlib`]
+(or one of its https://bintray.com/bintray/jcenter/org.jetbrains.kotlin%3Akotlin-stdlib-jre7[`kotlin-stdlib-jre7`]
+or https://bintray.com/bintray/jcenter/org.jetbrains.kotlin%3Akotlin-stdlib-jre8[`kotlin-stdlib-jre8`] variants).
+
+[[kotlin-extensions]]
+== Extensions
+
+[WARNING]
+====
+As of `Dysprosium-M1` (ie. `reactor-core 3.3.0.M1`), Kotlin extensions are deprecated in
```

Comment:
We should maybe be careful with the wording, and say instead:

As of `Dysprosium-M1` (ie. `reactor-core 3.3.0.M1`), Kotlin extensions are moved to a dedicated https://github.com/reactor/reactor-kotlin-extensions module . The package names start with `reactor.kotlin` instead of simply `reactor`.